code/src/java/pcgen/core/PCClass.java

```diff
@@ -434,10 +434,9 @@ public void setLevel(final int newLevel, final PlayerCharacter aPC)
 			SubstitutionClassApplication.checkForSubstitutionClass(this, newLevel, aPC);
 		}
 
-		for (PCClass pcClass : aPC.getClassSet())
-		{
-			aPC.calculateKnownSpellsForClassLevel(this);
-		}
+		aPC.getClassSet().stream()
```

Comment:
This feels like it's hiding a bug.  ignored -> this??

Reply:
I think the map can be removed?

Reply:
oh, you're right.